.travis.yml

```diff
@@ -1,62 +1,67 @@
-language: haskell
+# Based on https://docs.haskellstack.org/en/stable/travis_ci/
+#
+# Copy these contents into the root directory of your Github project in a file
+# named .travis.yml
+
+# Use new container infrastructure to enable caching
+sudo: false
+
+# Do not choose a language; we provide our own build tools.
+language: generic
+
+# Caching so the next build will be fast too.
+cache:
+  directories:
+  - $HOME/.stack
+
+# Ensure necessary system libraries are present
+addons:
+  apt:
+    packages:
+      - libgmp-dev
+      - libffi-dev
+      - libtinfo-dev
+      - zlib1g-dev
+      - pkg-config
+      - libgc-dev
 
 before_install:
-      ## Emacs 24.3
-      - sudo add-apt-repository -y ppa:cassou/emacs
-      - sudo apt-get update -qq
-      - sudo apt-get -f install
-      - sudo apt-get install -qq emacs24 emacs24-el
-
-      ## Emacs 24.4
-      - wget http://ftpmirror.gnu.org/emacs/emacs-24.4.tar.xz
-      - tar -xf emacs-24.4.tar.xz
-      - cd emacs-24.4
-      - ./configure --prefix=/usr/local/emacs-24.4 --with-xpm=no --with-gif=no
-      - make
-      - sudo make install
-      - cd ..
-
-      ## Idris prereqs
-      - sudo add-apt-repository --yes ppa:h-rayflood/llvm
-      - sudo apt-get update -qq
-      - sudo apt-get install -qq libgc-dev llvm-3.3
-      - sudo apt-get install -qq libghc-unordered-containers-dev libghc-mtl-dev libghc-network-dev libghc-xml-dev libghc-transformers-dev libghc-text-dev libghc-utf8-string-dev libghc-vector-dev libghc-split-dev libghc-ansi-terminal-dev libghc-ansi-wl-pprint-dev
-      # trifecta dependencies
-      - sudo apt-get install -qq libghc-blaze-builder-dev libghc-blaze-html-dev libghc-bifunctors-dev libghc-hashable-dev libghc-semigroups-dev libghc-semigroupoids-dev libghc-parallel-dev libghc-comonad-dev libghc-terminfo-dev libghc-keys-dev
-      # Haddock dependencies
-      - sudo apt-get install hscolour
-      # test dependency
-      - sudo apt-get install -qq expect
-      - cabal install alex
+  # Download and unpack the stack executable
+  - mkdir -p ~/.local/bin
+  - export PATH=$HOME/.local/bin:$PATH
+  - travis_retry curl -L https://www.stackage.org/stack/linux-x86_64 | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
+
+  ## Emacs 24.3
+  - sudo add-apt-repository -y ppa:cassou/emacs
+  - sudo apt-get update -qq
+  - sudo apt-get -f install
+  - sudo apt-get install -qq emacs24 emacs24-el
+
+  ## Emacs 24.4
+  - wget http://ftpmirror.gnu.org/emacs/emacs-24.4.tar.xz
+  - tar -xf emacs-24.4.tar.xz
+  - cd emacs-24.4
+  - ./configure --prefix=/usr/local/emacs-24.4 --with-xpm=no --with-gif=no
+  - make
+  - sudo make install
+  - cd ..
```

Comment:
Shouldn't we now test on Emacs 25.3 and 24.5, and cache the resulting directories?

Reply:
Yes, but I was planning to do that in a later step. Actually I would like to have a clear separation of builds with a "matrix" of emacs versions versus idris versions...